src/internal/storage/server.go

```
@@ -220,23 +225,48 @@ func (w *writer) Write(data []byte) (int, error) {
 
 func (s *Server) ReadFilesetCDR(request *storage.ReadFilesetRequest, server storage.Fileset_ReadFilesetCDRServer) error {
 	ctx := server.Context()
-	// TODO: Use task chain here.
-	return s.readFileset(ctx, request, func(f fileset.File) error {
+	cache, err := lru.New[string, string](128)
+	if err != nil {
+		panic(err)
```

lets just return the error here.